google_guest_agent/addresses.go

```diff
@@ -532,8 +532,22 @@ func enableNetworkInterfaces() error {
 }
 
 func isNetworkManagerRunning() bool {
-	res := runCmd(exec.Command("systemctl", "is-active", "--quiet", "NetworkManager"))
-	return res == nil
+	// SystemD
+	if systemctl, err := exec.LookPath("systemctl"); err == nil {
+		return runCmd(exec.Command(systemctl, "is-active", "--quiet", "NetworkManager")) == nil
+	}
+	// Upstart
+	if runCmd(exec.Command("/sbin/init", "--version")) == nil {
+		outString :=runCmdOutput(exec.Command("status", "network-manager")).out
+		return strings.Contains(outString, "running")
+	}
+	// Sysvinit
+	if _, err := os.Stat("/etc/init.d/network"); err == nil {
+		return runCmd(exec.Command("service  ", "network", "status")) == nil
```

Comment:
i don't think asking for a 'network' service necessarily implies NetworkManager is running. is there maybe a mechanism NetworkManager itself provides? on my gLinux system, i see /var/run/NetworkManager exists, if we can confirm that is uniformly true, maybe it will simplify this logic?

Reply:
i thought "network" and "networkManager" are refer to the same thing. But obviously that's not true. it should just named network-manager
/etc/init.d/network-manager